src/uct/cuda/base/cuda_md.c

```diff
@@ -280,28 +281,42 @@ uct_cuda_base_query_md_resources(uct_component_t *component,
     ucs_status_t status;
     char device_name[10];
     int num_gpus;
+    ucs_sys_bus_id_t bus_id;
 
     cudaErr = cudaGetDeviceCount(&num_gpus);
     if ((cudaErr != cudaSuccess) || (num_gpus == 0)) {
         return uct_md_query_empty_md_resource(resources_p, num_resources_p);
     }
 
-    for (cuda_device = 0; cuda_device < num_gpus; ++cuda_device) {
-        status = uct_cuda_base_get_sys_dev(cuda_device, &sys_dev);
-        if (status == UCS_OK) {
-            ucs_snprintf_safe(device_name, sizeof(device_name), "GPU%d",
-                              cuda_device);
-            status = ucs_topo_sys_device_set_name(sys_dev, device_name);
-            ucs_assert_always(status == UCS_OK);
+    ucs_spin_lock(&uct_cuda_base_lock);
```

Comment:
is it needed to protect `uct_cuda_sys_dev_bus_id_map`? If yes, when can it be accessed concurrently?

Reply:
As this logic is called as part of md_query functionality and because we access global structure uct_cuda_sys_dev_bus_id_map, I used spin_lock around the access. Is this unnecessary? Is it guaranteed that query_md_resources will be called by only thread at any given time?

Reply:
Currently we support multi-threading for UCP only, and all UCP API calls (including progress) are protected by a global lock.
Imo, there is not need for lock here then.